shell/platform/windows/text_input_plugin.cc

```diff
@@ -140,21 +177,38 @@ void TextInputPlugin::ComposeEndHook() {
   if (active_model_ == nullptr) {
     return;
   }
+  std::string text_before_change = active_model_->GetText();
+  TextRange selection_before_change = active_model_->selection();
   active_model_->CommitComposing();
   active_model_->EndComposing();
-  SendStateUpdate(*active_model_);
+  if (enable_delta_model) {
+    std::string text = active_model_->GetText();
+    TextEditingDelta delta = TextEditingDelta(text);
+    SendStateUpdateWithDelta(*active_model_, &delta);
+  } else {
+    SendStateUpdate(*active_model_);
+  }
 }
 
 void TextInputPlugin::ComposeChangeHook(const std::u16string& text,
```

Comment:
@Renzo-Olivares I tried out the composing stuff by typing Chinese into the [delta-text-editor-demo](https://github.com/justinmc/delta-text-editor-demo).  The demo seemed to be broken in how it handled replacements, as in when a composing region gets replaced by a final character or characters.  I updated it so it worked for most cases, but sometimes it would end up inserting black text that should have been colored.  We should figure out what to do with that demo before our versions diverge too much.